libdnf/dnf-sack.cpp

```diff
@@ -239,7 +239,7 @@ try_to_use_cached_solvfile(const char *path, Repo *repo, int flags, const unsign
         }
         return FALSE;
     }
-    std::unique_ptr<SolvUserdata> solv_userdata = solv_userdata_read(fp_cache);
+    std::unique_ptr<SolvUserdata, decltype(free)*> solv_userdata = solv_userdata_read(fp_cache);
```

Comment:
The `solv_userdata` are allocated by `libsolv` which uses `solv_malloc`. `solv_malloc` does use `malloc` internally but that is an implementation detail of `libsolv`.
Therefore I think we should use `solv_free` instead of `free` for the `decltype` all through out the PR.

Reply:
Good call.  PR has been updated.